UnityProject/Assets/Scripts/Core/SecureStuff/HubValidation.cs

```diff
@@ -104,21 +127,36 @@ private static void SaveCashedURLConfiguration(URLData URLData)
 
 		}
 
-		private static void AddTrustedHost(string Host, bool IsAPI)
+		private static void AddTrustedHost(Uri URL, bool IsAPI)
 		{
 			if (IsAPI)
 			{
-				AllowedAPIHosts.Add(Host);
+				if (URL.Host == "raw.githubusercontent.com")
+				{
+					var StringURL = URL.ToString();
+
+					var splits = StringURL.Split("/");
+					if (splits.Length < 6)
+					{
+						return;
+					}
+					AllowedGithubRepositories.Add(splits[3].ToLower() + "/" + splits[4].ToLower() + "/" + splits[5].ToLower());
```

Comment:
It is probably better to use the segments of the url, instead of manually splitting the string. I would also advise to name the segments so people who are reading the code can follow it without hassle.

```cs
private static void AddTrustedHost(Uri url, bool isAPI)
{
    if (isAPI)
    {
        if (url.Host == "raw.githubusercontent.com")
        {
            var segments = url.Segments;

            // Expected format: /username/reponame/branchname/...
            if (segments.Length >= 4) 
            {
                var username = segments[1].TrimEnd('/');
                var repoName = segments[2].TrimEnd('/');
                var branchName = segments[3].TrimEnd('/');
                
                AllowedGithubRepositories.Add($"{username.ToLower()}/{repoName.ToLower()}/{branchName.ToLower()}");
            }
        }
        else
        {
            AllowedAPIHosts.Add(url.Host);
        }
    }
    else
    {
        AllowedOpenHosts.Add(url.Host);
    }

    SaveCashedURLConfiguration(new URLData()
    {
        AllowedAPIHosts = AllowedAPIHosts,
        AllowedOpenHosts = AllowedOpenHosts,
        AllowedGithubRepositories = AllowedGithubRepositories
    });
}
```